util/src/testing.rs

```diff
@@ -0,0 +1,95 @@
+use core::fmt;
+use core::mem;
+use core::slice;
+
+/// Internal definition type used for a test.
+pub struct Test {
```

Comment:
nit/tioli: since this has all pub fields, should there maybe be a private `_p: ()` or something to stop it from being constructed from anywhere?

Reply:
It has to be constructed from within the macro, which is why it has all-pub fields, so I can't add a `_p: ()` or the `decl_test!` macro won't work.

It's technically harmless to construct it anywhere you like, it won't do anything unless it's in the correct section.

Reply:
ah, nvm, i missed that! seems like it could have a const-fn ctor but that might just be boilerplate?

Reply:
Yeah, I think that might just end up being boilerplate. I'm inclined to just leave it as-is.